meta-iot2000-example/recipes-tools/switchmode/files/switchmode.c

```diff
@@ -59,6 +71,22 @@ static int set_mode(int file, char *device, char *mode)
 		return 1;
 	}
 
+	if (option) {
+		if (strcmp(option, "-t") == 0 ||
+		    strcmp(option, "--terminate") == 0) {
+			if (!(rs485conf.flags & SER_RS485_ENABLED)) {
+				fprintf(stderr,
+					"Termination not supported in RS232 "
+					"mode\n");
+				return 1;
```

Comment:
This can be considered an argument error as well and consequently may have a different exit code.